server-common/src/test/java/org/apache/kafka/server/common/FeaturesTest.java

```diff
@@ -115,8 +115,8 @@ public void testDefaultValueAllFeatures(Features feature) {
 
     @ParameterizedTest
     @EnumSource(Features.class)
-    public void testLatestProductionMapsToLatestMetadataVersion(Features features) {
-        assertEquals(features.latestProduction(), features.defaultValue(MetadataVersion.LATEST_PRODUCTION));
+    public void testLatestProductionIsNotBehindLatestMetadataVersion(Features features) {
```

Comment:
We already validate this in Features. Is this test still needed?

Reply:
It may be worth confirming this fails just to confirm? In case the code gets changed?

Reply:
We can keep this test. But this test only covers one of the three conditions in `Features.validateDefaultValueAndLatestProductionValue`. If we are worried about code changes, should we have tests covering other conditions too?